src/NumericalAlgorithms/FiniteDifference/AoWeno.hpp

```diff
@@ -0,0 +1,334 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <utility>
+
+#include "NumericalAlgorithms/FiniteDifference/Reconstruct.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Math.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t Dim>
+class Direction;
+template <size_t Dim>
+class Index;
+/// \endcond
+
+namespace fd::reconstruction {
+namespace detail {
+template <size_t NonlinearWeightExponent>
+struct Aoweno53Reconstructor {
```

Comment:
I suggest using `AoWeno53Reconstructor` (or changing the file names to Aoweno) for consistency